src/WebJobs.Script/azurefunctions/functions.js

```diff
@@ -127,3 +128,27 @@ function getEntryPoint(f, context) {
 
     return f;
 }
+
+function stripFunctions(val) {
+    return stripFunctionsDepth(val, 2);
+}
+
+// strip functions from 'val' nested up to 'depth' within arrays and objects
+// ignores non-objects and buffers
+function stripFunctionsDepth(val, depth) {
```

Comment:
Interestingly enough this doesn't work. No matter what you return from CanConvert, json.net will keep recursively calling CanConvert on internal properties until it hits some v8::nan properties and throws up. Looking at overriding DefaultContractResolver to see if I can ignore func type properties from there on deserialize